torch/_meta_registrations.py

```diff
@@ -529,3 +525,56 @@ def meta_embedding_bag_forward_only(weight, indices, offsets, *args):
     if offsets.device.type == "cpu":
         bag_size = offsets.new_empty(offsets.size())
     return output, offset2bag, bag_size, max_indices
+
+
+def _get_reduction_dtype(input, dtype, promote_int_to_long=True):
+    # if specified, dtype takes precedence
+    if dtype:
+        return dtype
+
+    if input.dtype.is_floating_point or input.dtype.is_complex:
+        return input.dtype
+    elif promote_int_to_long:
+        return torch.long
+
+    return input.dtype
+
+
+def _get_reduction_output_size(input, dims, keepdim):
```

Comment:
can we use `compute_reduction_output_shape` (See `meta_var_mean_correction`)?

Reply:
Updated. Thanks!